src/test/scala/chiseltest/formal/MagicPacketTrackerTests.scala

```diff
@@ -0,0 +1,383 @@
+  }
+
+  it should "find bug in QueueV6 w/ pipe = false" taggedAs FormalTag in {
+    assume(DefaultBackend != CVC4EngineAnnotation, "CVC4 is too slow with some of these tests (5h+)")
+    val e = intercept[FailedBoundedCheckException] {
+      verify(new QueueFormalTest(new MyQueueV6(DefaultDepth, 32, pipe = false, fixed = false)), defaultOptions)
+    }
+    assert(e.failAt == 3)
+  }
+
+  it should "verify fix for QueueV6 w/ pipe = false" taggedAs FormalTag in {
+    assume(DefaultBackend != CVC4EngineAnnotation, "CVC4 is too slow with some of these tests (5h+)")
+    verify(new QueueFormalTest(new MyQueueV6(DefaultDepth, 32, pipe = false, fixed = true)), defaultOptions)
+  }
+
+  it should "find bug in QueueV6 w/ pipe = true" taggedAs FormalTag in {
+    assume(DefaultBackend != CVC4EngineAnnotation, "CVC4 is too slow with some of these tests (5h+)")
+    val e = intercept[FailedBoundedCheckException] {
+      verify(new QueueFormalTest(new MyQueueV6(DefaultDepth, 32, pipe = true, fixed = false)), defaultOptions)
+    }
+    assert(e.failAt == 2)
+  }
+
+  it should "verify fix for QueueV6 w/ pipe = true" taggedAs FormalTag in {
+    assume(DefaultBackend != CVC4EngineAnnotation, "CVC4 is too slow with some of these tests (5h+)")
+    verify(new QueueFormalTest(new MyQueueV6(DefaultDepth, 32, pipe = true, fixed = true)), defaultOptions)
+  }
+
+  it should "verify some versions of the Queue from the Chisel standard library" taggedAs FormalTag in {
+    assume(DefaultBackend != CVC4EngineAnnotation, "CVC4 is too slow with some of these tests (5h+)")
+    verify(new ChiselQueueTest(DefaultDepth, 32, false, false), defaultOptions)
+    verify(new ChiselQueueTest(DefaultDepth, 32, false, true), defaultOptions)
+    verify(new ChiselQueueTest(DefaultDepth, 32, true, false), defaultOptions)
+    verify(new ChiselQueueTest(DefaultDepth, 32, true, true), defaultOptions)
+  }
+}
+
+private class PacketTrackerCounterOverflowTest extends Module {
+  // our queue can take up to 8 elements
+  val dut = Module(new Queue(UInt(32.W), 8))
+  val io = IO(chiselTypeOf(dut.io)) ; io <> dut.io
+  // however for some reason we misconfigure the depth of the tracker
+  MagicPacketTracker(enq = dut.io.enq, deq = dut.io.deq, depth = 3)
+}
+
+
+// very similar to the QueueSpec in the chisel3 tests, but a formal test
+private class ChiselQueueTest(queueDepth: Int, bitWidth: Int, useSyncReadMem: Boolean, hasFlush: Boolean) extends Module {
+  val dut = Module(new Queue(UInt(bitWidth.W), queueDepth, useSyncReadMem = useSyncReadMem, hasFlush = hasFlush))
+  val io = IO(chiselTypeOf(dut.io)) ; io <> dut.io
+  // we need to tie the flush pin to false, because the MagicPacketTracker cannot (currently)
+  // handle flushes, it will complain about a missing packet if a flush happens
+  dut.io.flush.foreach(_ := false.B)
+  MagicPacketTracker(enq = dut.io.enq, deq = dut.io.deq, depth = queueDepth)
```

Comment:
I see that the MagicPacketTracker apply function creates an Input for startTracking in the context in which its called, but this is unlike other Chisel modules whose apply functions gives you a handle to the instantiated module and you have to manually wire its IO upwards... I like your API better though since it avoids that wiring boilerplate.